pkg/connectors/clients/policymanager_grpc.go

```diff
@@ -39,11 +46,327 @@ func NewGrpcPolicyManager(name string, connectionURL string, connectionTimeout t
 	}, nil
 }
 
-func (m *grpcPolicyManager) GetPoliciesDecisions(ctx context.Context, in *pb.ApplicationContext) (*pb.PoliciesDecisions, error) {
-	result, err := m.client.GetPoliciesDecisions(ctx, in)
-	return result, errors.Wrap(err, fmt.Sprintf("get policies decisions from %s failed", m.name))
+func (m *grpcPolicyManager) GetPoliciesDecisions(
+	in *openapiclientmodels.PolicyManagerRequest, creds string) (*openapiclientmodels.PolicyManagerResponse, error) {
+	log.Println("openapiclientmodels.PolicyManagerRequest: received in GetPoliciesDecisions: ", *in)
+	appContext, _ := ConvertOpenAPIReqToGrpcReq(in, creds)
+	log.Println("appContext: created from convertOpenApiReqToGrpcReq: ", appContext)
+
+	result, _ := m.client.GetPoliciesDecisions(context.Background(), appContext)
+
+	log.Println("GRPC result returned from GetPoliciesDecisions:", result)
+	policyManagerResp, _ := ConvertGrpcRespToOpenAPIResp(result)
+
+	res, err := json.MarshalIndent(policyManagerResp, "", "\t")
+	log.Println("err :", err)
+	log.Println("policyManagerResp: created from convGrpcRespToOpenApiResp")
+	log.Println("marshalled response:", string(res))
+	return policyManagerResp, nil
 }
 
 func (m *grpcPolicyManager) Close() error {
 	return m.connection.Close()
 }
+
+// ref: https://sosedoff.com/2014/12/15/generate-random-hex-string-in-go.html
+func randomHex(n int) (string, error) {
+	bytes := make([]byte, n)
+	if _, err := rand.Read(bytes); err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(bytes), nil
+}
```

Comment:
Functions that might be generally useful should go to their own package in `pkg/`

Reply:
Added a `utils` folder in `pkg/`. Please review. Thanks.